packages/client/src/pages/my-antennas/editor.vue

```diff
@@ -44,135 +44,110 @@
-			}
-		}
-	},
+	if (src === 'group' && userGroups === null) {
+		const groups1 = await os.api('users/groups/owned');
+		const groups2 = await os.api('users/groups/joined');
 
-	created() {
-		this.name = this.antenna.name;
-		this.src = this.antenna.src;
-		this.userListId = this.antenna.userListId;
-		this.userGroupId = this.antenna.userGroupId;
-		this.users = this.antenna.users.join('\n');
-		this.keywords = this.antenna.keywords.map(x => x.join(' ')).join('\n');
-		this.excludeKeywords = this.antenna.excludeKeywords.map(x => x.join(' ')).join('\n');
-		this.caseSensitive = this.antenna.caseSensitive;
-		this.withReplies = this.antenna.withReplies;
-		this.withFile = this.antenna.withFile;
-		this.notify = this.antenna.notify;
-	},
+		userGroups = [...groups1, ...groups2];
+	}
+});
 
-	methods: {
-		async saveAntenna() {
-			if (this.antenna.id == null) {
-				await os.apiWithDialog('antennas/create', {
-					name: this.name,
-					src: this.src,
-					userListId: this.userListId,
-					userGroupId: this.userGroupId,
-					withReplies: this.withReplies,
-					withFile: this.withFile,
-					notify: this.notify,
-					caseSensitive: this.caseSensitive,
-					users: this.users.trim().split('\n').map(x => x.trim()),
-					keywords: this.keywords.trim().split('\n').map(x => x.trim().split(' ')),
-					excludeKeywords: this.excludeKeywords.trim().split('\n').map(x => x.trim().split(' ')),
-				});
-				this.$emit('created');
-			} else {
-				await os.apiWithDialog('antennas/update', {
-					antennaId: this.antenna.id,
-					name: this.name,
-					src: this.src,
-					userListId: this.userListId,
-					userGroupId: this.userGroupId,
-					withReplies: this.withReplies,
-					withFile: this.withFile,
-					notify: this.notify,
-					caseSensitive: this.caseSensitive,
-					users: this.users.trim().split('\n').map(x => x.trim()),
-					keywords: this.keywords.trim().split('\n').map(x => x.trim().split(' ')),
-					excludeKeywords: this.excludeKeywords.trim().split('\n').map(x => x.trim().split(' ')),
-				});
-				this.$emit('updated');
-			}
-		},
+async function saveAntenna() {
+	if (props.antenna.id == null) {
+		await os.apiWithDialog('antennas/create', {
+			name,
+			src,
+			userListId,
+			userGroupId,
+			withReplies,
+			withFile,
+			notify,
+			caseSensitive,
+			users: users.trim().split('\n').map(x => x.trim()),
+			keywords: keywords.trim().split('\n').map(x => x.trim().split(' ')),
+			excludeKeywords: excludeKeywords.trim().split('\n').map(x => x.trim().split(' ')),
+		});
+		emit('created');
+	} else {
+		await os.apiWithDialog('antennas/update', {
+			antennaId: props.antenna.id,
+			name,
+			src,
+			userListId,
+			userGroupId,
+			withReplies,
+			withFile,
+			notify,
+			caseSensitive,
+			users: users.trim().split('\n').map(x => x.trim()),
+			keywords: keywords.trim().split('\n').map(x => x.trim().split(' ')),
+			excludeKeywords: excludeKeywords.trim().split('\n').map(x => x.trim().split(' ')),
+		});
```

Comment:
This is a lot of code duplication.
```suggestion
	const data = {
		name,
		src,
		userListId,
		userGroupId,
		withReplies,
		withFile,
		notify,
		caseSensitive,
		users: users.trim().split('\n').map(x => x.trim()),
		keywords: keywords.trim().split('\n').map(x => x.trim().split(' ')),
		excludeKeywords: excludeKeywords.trim().split('\n').map(x => x.trim().split(' ')),
	};
	if (props.antenna.id == null) {
		await os.apiWithDialog('antennas/create', data);
		emit('created');
	} else {
		await os.apiWithDialog('antennas/update', data);
```

Reply:
I didn't apply your patch right here, because I needed to make some adjustments:
* I didn't know you made your "I'll have to add more lints" threat true already (`data` is not allowed 😛)
* On updating `antennaId` is required, this was removed in your deduplication